client/lib/fifo/fifo_windows.go

```diff
@@ -0,0 +1,79 @@
+package fifo
+
+import (
+	"io"
+	"net"
+	"os"
+	"sync"
+	"time"
+
+	winio "github.com/Microsoft/go-winio"
+)
+
+// PipeBufferSize is the size of the input and output buffers for the windows
+// named pipe
+const PipeBufferSize = int32(^uint16(0))
+
+type winFIFO struct {
+	listener net.Listener
+	conn     net.Conn
+	connLock sync.Mutex
+}
+
+func (f *winFIFO) Read(p []byte) (n int, err error) {
+	f.connLock.Lock()
+	defer f.connLock.Unlock()
+	if f.conn == nil {
+		c, err := f.listener.Accept()
+		if err != nil {
+			return 0, err
+		}
+
+		f.conn = c
+	}
+
+	// If the connection is closed then we need to close the listener
+	// to emulate unix fifo behavior
+	n, err = f.conn.Read(p)
+	if err == io.EOF {
+		f.listener.Close()
+	}
+	return n, err
+}
+
+func (f *winFIFO) Close() error {
+	return f.listener.Close()
+}
+
+// New creates a fifo at the given path and returns a reader for it. The fifo
+// must not already exist
+func New(path string) (io.ReadCloser, error) {
+	l, err := winio.ListenPipe(path, &winio.PipeConfig{
+		InputBufferSize:  PipeBufferSize,
+		OutputBufferSize: PipeBufferSize,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return &winFIFO{
+		listener: l,
+	}, nil
+}
+
+// OpenWriter opens a fifo that already exists and returns a writer for it
+func OpenWriter(path string) (io.WriteCloser, error) {
```

Comment:
Why not just open and return a readwritecloser?

Reply:
Haha this is what I initially had but @schmichael thought having seperate funcs for Reader/Writer would be beneficial. 

I settled on only having a writer here since `New` returns a reader. For our use, the data will be unidirectional so `New` would be the only place we want a reader.

This can totally be a ReaderWriterCloser, it just didn't make sense to me to ever need to read from the pipe (other than for the caller of `New`).

Reply:
Won't bike shed on this since it will do what we need but I would have both open/new return readwriteclosers